src/components/ModalPayment/ModalMegaGam/ModalMegaGam.tsx

```diff
@@ -0,0 +1,373 @@
+                {t('buyMeal.totalLabel')}{' '}
+                <b>{formatCurrency(donationAmountDollars * 100, 0)}</b>
+              </label>
+            </div>
+          </SelectMealModule>
+          <FiscalSponsorContainer>
+            {campaignState.nonprofit_id && (
+              <FiscalSponsor nonprofitId={campaignState.nonprofit_id} />
+            )}
+          </FiscalSponsorContainer>
+        </MealContainer>
+      </AmountContainer>
+
+      <FeesAndOtherContainer>
+        <FeesAndOther>
+          {' '}
+          <b>{t('paymentProcessing.amount.your_donation')}</b>
+          <span>{formatCurrency(donationAmountDollars * 100)}</span>
+        </FeesAndOther>
+        {matchAmount > 0 && (
+          <>
+            <FeesAndOther>
+              <b>{t('paymentProcessing.amount.donation_match')}</b>
+              <span>{formatCurrency(matchAmount * 100)}</span>
+            </FeesAndOther>
+          </>
+        )}
+        {squareFee && (
+          <FeesAndOther>
+            <span>
+              <b>{'Credit card processing fees'}</b>
+              <Tooltip
+                title={t('paymentProcessing.amount.feesTooltip').toString()}
+                placement="right"
+              >
+                <Help style={{ color: '#A6192E', fontSize: '1rem' }} />
+              </Tooltip>
+            </span>
+            <span>{formatCurrency(feesAmount)}</span>
+          </FeesAndOther>
+        )}
+      </FeesAndOtherContainer>
+      {matchAmount > 0 && (
+        <>
+          <FeesAndOther>
+            <b>{t('paymentProcessing.amount.total_donation_plus_match')}</b>
+            <span>
+              {matchAmount > 0 && formatCurrency(totalContribution * 100)}
+            </span>
+          </FeesAndOther>
+          <Divider />
+        </>
+      )}
+
+      {/* TODO (billy-yuan): Confirm whether Mega Gam campaigns in the future will also do donation matching */}
+      <TotalContainer>
+        <div>
+          {t('paymentProcessing.amount.donation_total')}:{' '}
+          <span>
+            {formatCurrency(donationAmountDollars * 100 + feesAmount)}
+          </span>{' '}
+          <br />
+        </div>
+      </TotalContainer>
+
+      <NextButton
+        type="button"
+        className={'modalButton--filled'}
+        onClick={openModal}
+        disabled={
+          numberOfMeals < 1 ||
+          numberOfMeals > COST_LIMIT_DOLLARS / props.costPerMealInDollars
+        }
+      >
+        {t('paymentProcessing.amount.submit')}
+      </NextButton>
+    </MegaGamContainer>
+  );
+};
+
+export default Modal;
+
+const MegaGamContainer = styled.form`
+  height: 360px;
+  max-width: 960px;
+  margin: 0 auto;
+`;
+
+const Header = styled.div`
+  font-family: 'Open Sans', sans-serif;
+  font-size: 30px;
+  font-weight: 600;
+`;
+
+const SubHeader = styled.p`
+  margin-top: 16px;
+  margin-bottom: 40px;
+`;
+
+const AmountContainer = styled.div`
+  background-color: #f7f7f7;
```

Comment:
Do we have a constant for this?

Reply:
no but this is used throughout the front end, so we can update this across the site in another PR. Will make a trello card